gui/src/3D/controls/slider3D.ts

```diff
@@ -0,0 +1,193 @@
+import { Observable } from "babylonjs/Misc/observable";
+import { Vector3 } from "babylonjs/Maths/math.vector";
+import { TransformNode } from "babylonjs/Meshes/transformNode";
+import { Scene } from "babylonjs/scene";
+import { Mesh } from "babylonjs/Meshes/mesh";
+import { Control3D } from "./control3D";
+import { MeshBuilder } from "babylonjs/Meshes/meshBuilder";
+import { PointerDragBehavior } from "babylonjs/Behaviors/Meshes/pointerDragBehavior";
+import { Material } from "babylonjs/Materials/material";
+import { StandardMaterial } from "babylonjs/Materials/standardMaterial";
+import { Color3 } from "babylonjs/Maths/math.color";
+import { AbstractMesh } from "babylonjs/Meshes/index";
+
+export interface ISlider3DOptions {
+    minimum?: number,
+    maximum?: number,
+    value?: number
+};
+
+/**
+ * Class used to create a slider in 3D
+ */
+ export class Slider3D extends Control3D {
+    private _sliderBarMaterial: Material;
+    private _sliderThumbMaterial: Material;
+    private _sliderThumb: Mesh;
+    private _sliderBar: Mesh;
+
+    private _minimum: number;
+    private _maximum: number;
+    private _value: number;
+
+    /** Observable raised when the sldier value changes */
+    public onValueChangedObservable = new Observable<number>();
+
+    /**
+     * Creates a new slider
+     * @param name defines the control name
+     */
+     constructor(name?: string, options?: ISlider3DOptions) {
+        super(name);
+
+        this._minimum =  options?.minimum ? options.minimum : 0;
+        this._maximum = options?.maximum ? options.maximum : 100;
+        this.value = options?.value ? options.value : 50;       
+    }    
+
+    /** Gets or sets minimum value */
+    public get minimum(): number {
+        return this._minimum;
+    }
+
+    public set minimum(value: number) {
+        if (this._minimum === value) {
+            return;
+        }
+
+        this._minimum = Math.min(value, this._maximum);
+        this.value = this._value;
+    }
+
+     /** Gets or sets maximum value */
+     public get maximum(): number {
+         return this._maximum;
+     }
+ 
+     public set maximum(value: number) {
+         if (this._maximum === value) {
+             return;
+         }
+ 
+         this._maximum = Math.max(value, this._minimum);
+         this.value = this._value;
+     }
+
+    /** Gets or sets current value */
+    public get value(): number {

```

Comment:
nit: do we need separate comments for get vs set? Or will this comment auto resolve for both get and set in an ide?

Reply:
I followed the other 3D controls style. But I  am okay to put the comment for both setter and getter. Thanks

Reply:
I was thinking about this too. I think having a single comment is fine, since the documentation generated from this puts both the setter and getter into a single entry.

For example, Mesh.ts's isUnIndexed
https://doc.babylonjs.com/typedoc/classes/babylon.mesh#isunindexed

Reply:
It does look like the comment gets duplicated, though...

Reply:
Agreed with Rick, having a single for both setter and getter make sense.

Reply:
sounds good